### PR TITLE
fix: add term code to data and fix term capitalisation

### PR DESCRIPTION
**Pull request by WillKocur**
This adds the term code to trade_over_time endpoint, so that the code can be shown in the frontend of SAT in the legend items.

Also I think fixes the issue of the first letter of the terms not being capitalised.

Linked PR: https://github.com/unepwcmc/sustainability-assessment-tool/pull/56

NOTE: TdV needs this PR on prod before this goes to prod: https://github.com/unepwcmc/tradeplus/pull/93

### CODE REVIEW

**Review by lucacug — CHANGES_REQUESTED** [review]
A couple of comments here; I haven't checked for an alternative solution, but adding the term codes like this is affecting Tradeview. 
It might need some work on the TdV side of things to fix it leaving the implementation as it is here

File: lib/modules/trade/grouping/trade_plus_static.rb [hunk 1]
```diff
@@ -343,7 +344,7 @@ def sanitise_column_names
       next if attribute == 'year' || attribute.nil?
       name = attribute.include?('id') ? 'id' : attribute.include?('iso') ? 'iso2' : attribute.include?('code') ? 'code' : 'name'
       @sanitised_column_names << name
-      attribute = "INITCAP(#{attribute})" if attribute == 'term'
+      attribute = "INITCAP(#{attribute})" if attribute == 'term_en'
```
[the remark this answers is not in the document]
Comment (WillKocur) [reply lib/modules/trade/grouping/trade_plus_static.rb:347]: Ah of course... I can put the frontend fix back in that just capitalises there if that's preferred for now. What do you think?

File: lib/modules/trade/grouping/trade_plus_static.rb [hunk 2]
```diff
@@ -140,7 +141,7 @@ def self.grouping_attributes
 
   def self.localize_grouping_attributes
     {
-      terms: ["term_#{@locale}", 'term_id'],
+      terms: ["term_#{@locale}", 'term_id', 'term_code'],
```
[the remark this answers is not in the document]
Comment (WillKocur) [reply lib/modules/trade/grouping/trade_plus_static.rb:144]: @lucacug Oh good spot, I didn't expect this! Sorry, should have checked. Will fix TdV side I reckon as tradeview is currently doing this in a roundabout way anyway, so should be able to simplify it there a little 👍

File: lib/modules/trade/grouping/trade_plus_static.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucacug) [reply lib/modules/trade/grouping/trade_plus_static.rb:347]: I think is better to have this logic here in the BE

File: lib/modules/trade/grouping/trade_plus_static.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (WillKocur) [reply lib/modules/trade/grouping/trade_plus_static.rb:144]: @lucacug TdV PR here - will add to ticket too. https://github.com/unepwcmc/tradeplus/pull/93

**Comment by lucacug** [comment]
looks good, merging!

**Pull request merged**